### PR TITLE
Changing default transport for stackdriver exporter

### PR DESCRIPTION
**Pull request by aravinsiva**
Changing default transporter to async as sync leads to latency issues. In addition will allow GCP Cloud Run clients to leverage atexit abilities of async exporter. Also addresses the following issue:
https://github.com/census-instrumentation/opencensus-python/issues/927

(This issue is addressed in two PR's. One for setting the default trace processor and the other for adding atexit handlers in the metrics exporter).

### CODE REVIEW

**Comment by aabmass** [comment]
This LGTM but I will defer to someone who knows why the other option is the current default.

File: contrib/opencensus-ext-stackdriver/README.rst [hunk 1]
```diff
@@ -35,20 +35,19 @@ This example shows how to report the traces to Stackdriver Trace:
     pip install google-cloud-trace
     pipenv install google-cloud-trace
 
-By default, traces are exported synchronously, which introduces latency during
-your code's execution. To avoid blocking code execution, you can initialize
-your exporter to use a background thread.
+By default, traces are exported asynchronously, to avoid latency during
```
Comment (sjbrunst) [inline contrib/opencensus-ext-stackdriver/README.rst:38]: nit: avoid -> reduce

I doubt even the async exporter can completely avoid latency.

**Comment by lzchen** [comment]
> This LGTM but I will defer to someone who knows why the other option is the current default.

I think @c24t  would have more insight on this.

**Comment by c24t** [comment]
> This LGTM but I will defer to someone who knows why the other option is the current default.

The decision to use the sync transport was made all the way back in https://github.com/census-instrumentation/opencensus-python/pull/63. It looks like that PR added support for async/background transport, but didn't change the default. In the meantime we've started using async transport as the default for other exporters, including OT exporters. I think we can do the same here.

**Pull request merged**